test/Microsoft.IdentityModel.Tokens.Tests/CustomSecurityKeys.cs

```diff
@@ -47,13 +47,22 @@ public CustomRsaSecurityKey(int keySize, PrivateKeyStatus privateKeyStatus, RSAP
             _privateKeyStatus = privateKeyStatus;
         }
 
+        public CustomRsaSecurityKey(int keySize, PrivateKeyStatus privateKeyStatus, RSAParameters parameters, string InternalId)
+            : base(parameters)
+        {
+            _keySize = keySize;
+            _privateKeyStatus = privateKeyStatus;
+        }
+
 #pragma warning disable CS0672
         public override bool HasPrivateKey => true;
 #pragma warning restore CS0672
 
         public override PrivateKeyStatus PrivateKeyStatus => _privateKeyStatus;
 
         public override int KeySize => _keySize;
+
+        internal override string InternalId  => "";
```

Comment:
This property has been added to the test key for testing that SignatureProviders corresponding to keys with empty internal IDs are not cached.